enhancements/agent-installer/automated-workflow.md

```diff
@@ -0,0 +1,491 @@
+later enhancement.
+
+The `create image` command will need to take inputs similar to what
+the baremetal IPI installer or ZTP APIs use to describe the cluster
+(name, API IPs, etc.) and host-specific network settings (static IPs,
+VLANs, etc.). Using the existing file and custom resource types allows
+us to use the same version management techniques that we do for the
+APIs within a cluster.
+
+The `create image` command will produce a single image as output, to
+save storage space and reduce management complexity. The assisted
+installer service can already generate images with boot scripts that
+recognize which host is booting and apply host-specific network
+settings and roles (such as running different services) accordingly,
+and we will reuse that technique (and code).
+
+One host, designated as "node0", will be configured to run the
+assisted service and a client that uses the assisted service API to
+automate the deployment.
+
+All other hosts will run the assisted agent, configured to talk to the
+service on node0.
+
+When all of the images are booted, the client on node0 will use the
+assisted service to trigger the deployment of the cluster (including
+applicable pre-flight validations) without further interaction from
+the user.
+
+The `create image` command will produce as output credentials to
+access the cluster that is about to be created. This allows a client,
+such as a tool monitoring deployment progress, to use the cluster's
+API as soon as the API service is available.
+
+During deployment, the third-party orchestration tool will need a way
+to monitor progress and discover errors. The assisted service
+read-only API endpoints running on node0 will report that data. In the
+event of failure, a read-only endpoint in that service will be
+available to download any logs that have been collected.
+
+We will be taking advantage of the assisted service's ability to run
+the bootstrap services "in place" to avoid having a separate bootstrap
+VM. Bootstrap will run on node0, where the assisted service runs, to
+ensure that the service and its API are available for as long as
+possible. When bootstrapping completes and node0 reboots, progress
+monitoring will need to shift to the OpenShift APIs (`ClusterVersion`
+and `ClusterOperator`). We will extend the `openshift-install wait-for
+install-complete` command to recognize when the cluster being deployed
+uses the agent installer and to use both APIs for monitoring and
+[progress
+reporting](https://github.com/openshift/assisted-service/blob/master/docs/enhancements/installation-progress-bar.md)
+in that case, including reporting when steps like validation fail and
+the cluster deployment is never going to succeed. Users will still be
+able to write their own monitoring logic using the assisted service
+and OpenShift APIs directly, but they will have the option of using
+the command line tool.
+
+For the fully automated workflow, the steps a user might normally
```

Comment:
another gap here is how the assisted-controller (bad name, it's just a job that finally completes) behaves.
Currently it approves nodes (in case of none platfrom, or always, I dont recall, send logs, progress reporting....

Reply:
From my observation, the assisted-controller is deployed on the 2-node control-plane before the bootstrap node reboots to join the cluster. When the bootstrap node reboots, we lose assisted-service and assisted-controller is unable to update the service with progress or push logs to it. The cluster installation, however, continues to progress to completion. The assisted-controller continues to approve CSRs even when assisted-service can't be reached.

Reply:
I've clarified this statement to mention approving agents before installation starts, since that's what I was thinking of when I wrote it originally.